funcx_endpoint/funcx_endpoint/executors/high_throughput/executor.py

```diff
@@ -566,14 +571,26 @@ def _queue_management_worker(self):
                             logger.debug(f"[MTHREAD] task:{tid} items in queue: {self.results_passthrough.qsize()}")
                             continue
 
-                        task_fut = self.tasks.pop(tid)
+                        try:
+                            task_fut = self.tasks.pop(tid)
+                        except KeyError:
+                            # This is triggered when the result of a cancelled task is returned
+                            # We should log, and proceed.
+                            logger.warning(f"[MTHREAD] Task:{tid} duplicate result received. Ignoring.")
```

Comment:
This could also happen if the task future has not yet been put into self.tasks (if some race condition happens, in future fiddling) right? The KeyError exception doesn't mean that it definitely used to be there, and maybe the comments should reflect that.

Reply:
I'll clarify the message.